pkg/service/replay/replay.go

```diff
@@ -951,6 +970,170 @@ func (r *Replayer) RunApplication(ctx context.Context, appID uint64, opts models
 	return r.instrumentation.Run(ctx, appID, opts)
 }
 
+func (r *Replayer) GetTestSetConf(ctx context.Context, testSet string) (*models.TestSet, error) {
+	return r.TestSetConf.Read(ctx, testSet)
+}
+
+func (r *Replayer) Templatize(ctx context.Context, testSets []string) error {
+	if len(testSets) == 0 {
+		if len(r.config.Templatize.TestSets) == 0 {
+			allTestSets, err := r.GetAllTestSetIDs(ctx)
+			if err != nil {
+				r.logger.Error("failed to get the test sets", zap.Error(err))
+			} else {
+				testSets = allTestSets
+			}
+		} else {
+			testSets = r.config.Templatize.TestSets
+		}
+	}
+	for _, testSetID := range testSets {
+		testSet, err := r.TestSetConf.Read(ctx, testSetID)
+		if err != nil || testSet == nil {
+			utils.TemplatizedValues = map[string]interface{}{}
+		} else {
+			utils.TemplatizedValues = testSet.Template
+		}
+		tcs, err := r.testDB.GetTestCases(ctx, testSetID)
+		if err != nil {
+			utils.LogError(r.logger, err, "failed to get test cases")
+			return err
+		}
+		if len(tcs) == 0 {
+			r.logger.Warn("The test set is empty. Please record some testcases to templatize.", zap.String("testSet:", testSetID))
```

Comment:
You should continue when there is no test-case in the test-set. The flow should not proceed.